zipline/testing/fixtures.py

```diff
@@ -108,30 +115,136 @@ def enter_instance_context(self, context_manager):
 
     After init_class_fixtures has been called, `cls.asset_finder` is populated
-    with an AssetFinder. The default finder just a `tmp_asset_finder`
+    with an AssetFinder. The default finder just a `tmp_asset_finder` whose
+    arguments come from:
+
+      equities=cls.make_equities_info(),
+      futures=cls.make_futures_info(),
+      exchanges=cls.make_exchanges_info(),
+      root_symbols=cls.make_root_symbols_info(),
+
+    Each of these fields may be overridden with a dataframe to write.
 
     This behavior can be altered by overriding `make_asset_finder` as a class
     method.
     """
     @classmethod
+    def _make_info(cls):
+        return None
+
+    make_equities_info = _make_info
+    make_futures_info = _make_info
+    make_exchanges_info = _make_info
+    make_root_symbols_info = _make_info
+
+    del _make_info
+
+    @classmethod
     def make_asset_finder(cls):
-        return cls.enter_class_context(tmp_asset_finder())
+        return cls.enter_class_context(tmp_asset_finder(
+            equities=cls.equities_info,
+            futures=cls.futures_info,
+            exchanges=cls.exchanges_info,
+            root_symbols=cls.root_symbols_info,
+        ))
 
     @classmethod
     def init_class_fixtures(cls):
         super(WithAssetFinder, cls).init_class_fixtures()
+        cls.equities_info = cls.make_equities_info()
+        cls.futures_info = cls.make_futures_info()
+        cls.exchanges_info = cls.make_exchanges_info()
+        cls.root_symbols_info = cls.make_root_symbols_info()
         cls.asset_finder = cls.make_asset_finder()
 
 
+class WithTradingEnvironment(WithAssetFinder):
+    """
+    ZiplineTestCase mixin providing cls.env as a class-level fixture.
+
+    After init_class_fixtures has been called, `cls.env` is populated
+    with a trading environment whose `asset_finder` is the result of
+    `cls.make_asset_finder`.
+
+    This behavior can be altered by overriding `make_trading_environment` as a
+    class method.
+    """
+    @classmethod
+    def make_load_function(cls):
+        return None
+
+    @classmethod
+    def make_trading_environment(cls):
+        return TradingEnvironment(
+            load=cls.make_load_function(),
+            asset_db_path=cls.asset_finder.engine,
+        )
+
+    @classmethod
+    def init_class_fixtures(cls):
+        super(WithTradingEnvironment, cls).init_class_fixtures()
+        cls.env = cls.make_trading_environment()
+
+
+class WithSimParams(WithTradingEnvironment):
+    SIM_PARAMS_YEAR = 2006
+    SIM_PARAMS_START = None
+    SIM_PARAMS_END = None
+    SIM_PARAMS_CAPITAL_BASE = float("1.0e5")
+    SIM_PARAMS_NUM_DAYS = None
+    SIM_PARAMS_DATA_FREQUENCY = 'daily'
+    SIM_PARAMS_EMISSION_RATE = 'daily'
+
+    @classmethod
+    def init_class_fixtures(cls):
+        super(WithSimParams, cls).init_class_fixtures()
+        cls.sim_params = factory.create_simulation_parameters(
```

Comment:
I could see the argument to leaving create_simulation_parameters, but I'd put a concrete default here so it's clear what you're getting as a consumer of this.

Also, we don't pass `year` anywhere to create_simulation_parameters as far as I can tell, so I think we can just remove that.